[GRO-480] Add docs for SQL runner. (#92)
hashboard-hq/documentation: pages/docs/sql-runner.mdx

---
description: Hashboard's SQL Runner allows you to write arbitrary SQL to quickly query your data across sources.
---

import { Eye, MoreVertical } from "lucide-react";

# SQL Runner

Hashboard's SQL Runner allows you to write arbitrary SQL to quickly query your data across data connections.

> To open the runner, press `Cmd + K` (Mac) or `Ctrl + K` (Windows/Linux). Alternatively, you can click `More` in the nav bar, then click `SQL runner`.

Reviewer: ```suggestion
To open the runner, press `Cmd + J` (Mac) or `Ctrl + J` (Windows/Linux). Alternatively, you can click `More` in the nav bar, then click `SQL runner`.
```

> Once opened, select a data connection on the top left, then write SQL and click `Run` (or `Cmd + Enter`) see results. You can also click <div style={{display: "inline-flex", alignItems: "baseline"}}><Eye style={{height: 16, alignSelf: "center"}} /> <strong>Preview</strong></div> next to any table on the left side to directly view its contents.

Reviewer: ```suggestion
Once opened, select a data connection on the top left, then write SQL and click `Run` or `Cmd + Enter` to run the query. You can also click <div style={{display: "inline-flex", alignItems: "baseline"}}><Eye style={{height: 16, alignSelf: "center"}} /> <strong>Preview</strong></div> next to any table on the left side to directly view its contents.
```

> You can also pre-populate the SQL Runner with the underlying query backing any Hashboard exploration. To do so, open a saved exploration, click the advanced menu (<MoreVertical style={{height: 16, display: "inline"}} />) on the top right of the page, then click `Open in SQL runner`.

Reviewer: ```suggestion
You can also pre-populate the SQL Runner with the underlying query backing any Hashboard exploration. To pre-populate an exploration's query open a saved exploration, click the advanced menu (<MoreVertical style={{height: 16, display: "inline"}} />) on the top right of the page, then click `Open in SQL runner`.
```
